authorization/resource/resource.go

```diff
@@ -0,0 +1,223 @@
+	}
+	err := m.db.Create(model).Error
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"resource_id": model.ID,
+			"err":         err,
+		}, "unable to create the resource")
+		return errs.WithStack(err)
+	}
+	log.Info(ctx, map[string]interface{}{
+		"resource_id": model.ID,
+	}, "Resource created!")
+	return nil
+}
+
+// Lookup looks for an existing resource with the given ID
+func (m *GormResourceRepository) Lookup(ctx context.Context, id string) (*Resource, error) {
+	if id == "" {
+		return nil, errs.New("Cannot lookup resource with empty resource ID")
+	}
+	log.Debug(nil, nil, "Looking for resource with id=%s\n", id)
+	// bind the assignee to an existing resource
+	resource, err := m.First(ResourceFilterByID(id))
+	if err != nil {
+		return nil, errs.Wrapf(err, "failed to lookup resource by id '%s'", id)
+	}
+	// use existing resource
+	log.Debug(nil, nil, "Using existing resource with ID: %v", resource.ID)
+	return resource, nil
+}
+
+// Save modifies a single record.
+func (m *GormResourceRepository) Save(ctx context.Context, model *Resource) error {
+	defer goa.MeasureSince([]string{"goa", "db", "resource", "save"}, time.Now())
+
+	obj, err := m.Load(ctx, model.ID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"resource_id": model.ID,
+			"ctx":         ctx,
+			"err":         err,
+		}, "unable to update the resource")
+		return errs.WithStack(err)
+	}
+	err = m.db.Model(obj).Updates(model).Error
+
+	log.Debug(ctx, map[string]interface{}{
+		"resource_id": model.ID,
+	}, "Resource saved!")
+
+	return errs.WithStack(err)
+}
+
+// Delete removes a single record.
+func (m *GormResourceRepository) Delete(ctx context.Context, id string) error {
+	defer goa.MeasureSince([]string{"goa", "db", "resource", "delete"}, time.Now())
+
+	obj := Resource{ID: id}
+	db := m.db.Delete(obj)
+
+	if db.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"resource_id": id,
+			"err":         db.Error,
+		}, "unable to delete the resource")
+		return errs.WithStack(db.Error)
+	}
+	if db.RowsAffected == 0 {
+		return errors.NewNotFoundError("resource", id)
+	}
+
+	log.Debug(ctx, map[string]interface{}{
+		"identity_id": id,
+	}, "Identity deleted!")
+
+	return nil
+}
+
+// Query expose an open ended Query model
+func (m *GormResourceRepository) Query(funcs ...func(*gorm.DB) *gorm.DB) ([]Resource, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "resource", "query"}, time.Now())
+	var resources []Resource
+	err := m.db.Scopes(funcs...).Table(m.TableName()).Find(&resources).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return nil, errs.WithStack(err)
+	}
+	log.Debug(nil, map[string]interface{}{
+		"resource_query": resources,
+	}, "Resource query executed successfully!")
+
+	return resources, nil
+}
+
+// First returns the first Resource element that matches the given criteria
+func (m *GormResourceRepository) First(funcs ...func(*gorm.DB) *gorm.DB) (*Resource, error) {
```

Comment:
The same question as "Query". Is this First() method is going to be used? I would suggest to add only CRUD methods that we will need for sure right now. Create(), Load(), Save(), Delete() and others only if needed. If not sure right now then let's not add it. We can add it later if needed.